spaceship/lib/spaceship/upgrade_2fa_later_client.rb

```diff
@@ -0,0 +1,91 @@
+require_relative 'globals'
+require_relative 'tunes/tunes_client'
+
+module Spaceship
+  class Client
+    def try_upgrade_2fa_later(response)
+      if ENV['SPACESHIP_SKIP_2FA_UPGRADE'].nil?
+        puts("Skipping automatic bypass of 2FA upgrade")
+        return false
+      end
+
+      puts("This account is being prompted to upgrade to 2FA")
+      puts("Attempting to automatically bypass the upgrade until a later date")
+      puts("To disable this, remove SPACESHIP_SKIP_2FA_UPGRADE=1 environment variable")
+
+      # Get URL that requests a repair and gets the widget key
+      widget_key_location = response.headers['location']
+      uri    = URI.parse(widget_key_location)
+      params = CGI.parse(uri.query)
+
+      widget_key = params.dig('widgetKey', 0)
+      if widget_key.nil?
+        STDERR.puts("Couldn't find widgetKey to continue with requests")
+        return false
+      end
+
+      # Step 1 - Request repair
+      response_repair = request(:get) do |req|
+        req.url(widget_key_location)
+      end
+
+      # Step 2 - Request repair options
+      response_repair_options = request(:get) do |req|
+        req.url("https://appleid.apple.com/account/manage/repair/options")
+
+        req.headers['scnt'] = response_repair.headers['scnt']
+        req.headers['X-Apple-Id-Session-Id'] = response.headers['X-Apple-Id-Session-Id']
```

Comment:
@joshdholtz I implement a similar solution and in my case, I have to get the value of the `X-Apple-Id-Session-Id` header from the cookie named `aidsp` returns in the previous request. Be sure it works as you did it 😄 
Maybe they already made a change...

Reply:
I tried _really_ hard to not have to look in the cookies 🙃